### ISSUE TITLE
0.83.2 tickets, email not sent until status changed

### ISSUE THREAD

**Issue by CCoruble** [glpi-project/glpi#1980]
Hi,

Context :
Actually working on updating GLPI from 0.83.2 to 9.1.2.
The direct update of database always failed, so I used version 0.90.5 as a buffer.
(I updated my database to 0.90.5 then to 9.1.2 and it worked without data lose)
Now that my new glpi is up to date I still have "old tickets", by this I mean tickets that was created on the 0.83.2 version.

Issue :
My problem is that the email follow up is not working correctly on 0.83.2 created tickets.
If I had a followups or a task on tickets (a 0.83.2 created tickets) , the email won't be sent to people that have the email follow up activated.
The only way for email to be sent is that the ticket status change (Ex: from processing to resolved).
After the status is changed all the email that sould have been already sent will be sent at the same time. 
Moreover the task displayed on the mail is NOT the last created task (youngest), it is always the FIRST (the oldest).
But when adding a new task (and changing the status, otherwise no email will be sent), the youngest task will be displayed. It seems like the task of the ticket need to be updated in order to display the good (the oldest) task on the mail.

Server steup :
![2017-03-29 11_01_19-glpi - setup](https://cloud.githubusercontent.com/assets/24699373/24446885/6ed0fd2e-146f-11e7-8965-adcf3065dcc6.png)

After a lot of test it seems that even 9.1.2 tickets suffer this issue ...
I tried to explain my issue as clear as I could.

Regards.
CCoruble

**Comment by CCoruble** [comment]
Any advice / hint to guide me where should I look in the database and maybe correct this bug by my own.
Any date for the 9.1.3 release ?

Regards
CCoruble

**Comment by trasher** [comment]
I do not have any idea what could be wrong, sorry.

And no, there is no planned date for any release right now.

**Comment by CCoruble** [comment]
Hi,
Well ... that's a bad news.
The rules for email sending were transported from the 0.83.2 to 9.1.2, the problem may come from this. I looked at my DB and I saw that there is 27 data on the glpi_notifications.

![2017-03-29 15_03_58-10 2 3 34 _ localhost _ glpi _ glpi_notifications _ phpmyadmin 4 5 4 1deb2ubuntu](https://cloud.githubusercontent.com/assets/24699373/24455755/140aeb4e-1491-11e7-9097-c120a329804a.png)

Here is the new data from this table. That may not come from this one but trying my best, maybe if you know from which table this issue come from, I'll check how to fix it by my own.

**Comment by CCoruble** [comment]
Hi,
I got some bad news ...
I tried a new glpi on 9.1.2, so a fresh install wihtout any importation, the most pure installation I could make. 
And the bug is still happening ... this is a bug version not a migration bug ...
The followups / task mail aren't send until you update the status of the ticket or just go on "ticket" tab and clik "save" (you don't even need to modify something, just saving is enough).

**Comment by CCoruble** [comment]
Hi,
Test on 0.90.5 and I also got this bug ... there must be something wrong on my configuration / installation. 
I followed this tutorial : https://doc.ubuntu-fr.org/glpi
Tell me please if there is something wrong on their explanation to install GLPI. 
This might come from here, that's weird that nobody spotted this bug since 0.90.5, this must come from my installation ...
Please help !

regards

**Comment by CCoruble** [comment]
Hi, tested a new fresh installation and noticed a bug that happens only once, on the first time you connect (with glpi/glpi), and disappear after the first connection.

![2017-03-30 11_26_04-glpi - interface standard](https://cloud.githubusercontent.com/assets/24699373/24497488/defd7736-153b-11e7-80c6-6c40d6489a3e.png)

Regards

**Comment by ladenree76000** [comment]
HI,
have a look to mail queue.
it's a new behaviour different from 0.83 : 
followups notification (and some others) are sent to mail queue.
you can set delay after wich email are set ( your trigger is either a cron task either glpi activity)

**Comment by trasher** [comment]
Hello,

The first part is true: notifications are sent to the mail queue.

But the way it is actually processed differs... For all itil objects; they're sent immediately. For all other; a crontask is used.

If I'm correct, this change are from 0.85.

**Comment by CCoruble** [comment]
Hi, 
Got the result of all my test.
I test on fresh install of version 0.85.5 / 0.90.5 / 9.1.2 and this happened very times, based on the same installation process.
The creation / change status email were immediately send, but when I added a followup the mail wasn't send until the status changed.

I also looked at the mail queue. I only checked for 9.1.2.
The mail for add followup / task (that should be sent) are present in the mail queue, so I checked if the automatic action were correctly set. "queuedmail" is set be started every minuts and when you look at "next run" this is precised "(as soon as possible)". BUT the "queuedmail" automatic action never start unless I start it by myself.

Something weird : If there is a mail on the mail queue, an add followup notification for example, and I change the status of the ticket, all the mail in the queue mail are sent BUT the "queuedmail" automatic action WAS NOT activated !!!!
As said @trasher some mails are sent directely (tickets add / tickets status change I guess) but some goes to the mail queue, that should be run every minuts. The problem is that the "queuedmail" automatic action doesn't seems to work correctly. Moreover the mail that are directly sent, seems to affect the mail in the mail queue and force them to be sent WITHOUT running "queuedmail" automatic action.

Sorry, this post might be long but I wanna be sure you understand what I mend what my test showed.

Regards

**Comment by CCoruble** [comment]
Hi,
Just found there is 2 cron file "glpi_cron_curl.sh" & "glpi_cron_lynx.sh" on "glpi/scripts".
It is precised on both of them that we should edit the URL leading to "http://YOURSERVER/glpi/front/cron.php", could this have a link to "automatic action" bug ? 
Thoose 2 files are NOT mentionned on the official installation tutorial. Also changed some crontab lines and it worked !!!! 
Checking everything to be sure what makes the automatic action finally works and I'll tell you !

Regards

**Comment by CCoruble** [comment]
Hi,
the answer was pretty simple ... I feel sad so much hour of research for something simple.
Did not find out why it is buged but how tyo fix this.
Go to "automatic action"  and change the "run mode" from "GLPI" to "CLI"
Then open crontab with "crontab -e" and add this line :
*/1 * * * * /usr/bin/php /var/www/html/glpi/front/cron.php &>/dev/null
This will make the action works every minuts and so all the mail in the mail queue will be sent each minuts !

Tell me if you got more info about how you will fix the bug in later versions and where it come from, if it's a bug.

Regards

**Comment by ladenree76000** [comment]
this question was asked so much time on the forum that anybody thought it was bad configuration from automatic task.
look for "glpi cron mailqueue" on your favorite search engine you'll find many topics.

**Comment by trasher** [comment]
Cron taks can be executed using two modes:
- GLPI: the cron script will be called when a page is displayed (a page, not a tab change or something similar). So, nothing happens until someone loads a page.
- CLI: rely on system crontab you have to properly configure.

As far as I understand, there is no bug here; I'm closing.

**Comment by CCoruble** [comment]
Hi,
Sorry for posting on a closed post.
As far as I understand your explanation @trasher , what I described earlier did not correspond to what you said ?!

> GLPI: the cron script will be called when a page is displayed (a page, not a tab change or something similar). So, nothing happens until someone loads a page.

When the "queuedmail" automatic action was on GLPI run mode and set to run each minutes, loading pages DID NOT activate him... even with differents account and different page.
(I did login on different account and load a lot of pages)

Let's try to make it clear :
**Is what you call "loading a page" correspond to refreshing a ticket status ?** 
**If no :**
There is a problem ! Because refreshing the status of a ticket was the ONLY WAY for mail to be sent. Moreover not all the queuedmail were sent, only the one concerning the ticket that got his status refreshed. 

**If yes :**
What you said correspond to the actual behavior of GLPI, there is no bug. But why would this be the only way for mail, concerning a ticket followup, to be sent ?

I may not correctly understand your explanation, but I think is a real issue.
